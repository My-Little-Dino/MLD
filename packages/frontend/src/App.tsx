import React from 'react';
import { ToastContainer } from 'react-toastify';
import styled from 'styled-components';
import Header  from './components/Header';
import InfoContract from './components/InfoContract';

import 'react-toastify/dist/ReactToastify.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import './styles/globals.css';
import './styles/header.css';

const App = () => {
	return (
		<Wrapper>
			<Container>
				
				<Header />
				<Content>
					{/* <InfoContract /> */}
				</Content>
			</Container>
			<ToastContainer
				position="bottom-right"
				newestOnTop={false}
				pauseOnFocusLoss={false}
				pauseOnHover={false}
				rtl={false}
			/>
		</Wrapper>
	);
};

const Flex = styled.div`
	display: flex;
	align-items: center;
`;

const Topbar = styled.div`
	display: flex;
	justify-content: space-between;
	align-items: center;
	height: 74px;
	width: 100%;
`;

const Content = styled.div`
	display: flex;
	align-items: center;
	justify-content: center;
	flex: 1;
	width: 100%;
`;

const Container = styled.div`
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
	min-height: 100vh;
	padding: 0px 24px;
	max-width: 1200px;
	margin: 0 auto;
`;

const Wrapper = styled.div`
	background-color: #0093e9;
	background-image: linear-gradient(160deg, #0093e9 0%, #80d0c7 100%);
	font-size: 1rem;
	color: white;
`;

export default App;
