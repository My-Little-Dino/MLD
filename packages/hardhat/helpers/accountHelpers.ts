import { fromBech32 } from '@harmony-js/crypto';
import { isBech32Address } from '@harmony-js/utils';

export const parseOneAddress = (address: string): string => (isBech32Address(address) ? fromBech32(address) : address);
