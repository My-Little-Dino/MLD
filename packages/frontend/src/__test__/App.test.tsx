import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';

import App from '../App';

import { mockComponent } from '../helpers/testHelper';

jest.mock('../components/Account', () => (props: any) => mockComponent('Account', props));
jest.mock('../components/Balance', () => (props: any) => mockComponent('Balance', props));
jest.mock('../components/InfoContract', () => (props: any) => mockComponent('InfoContract', props));

describe('Test App component', () => {
	const component = <App />;

	const setup = () => render(component);

	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<App />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it('Should match snapshot', () => {
		const { container } = setup();
		expect(container).toMatchSnapshot();
	});
});
