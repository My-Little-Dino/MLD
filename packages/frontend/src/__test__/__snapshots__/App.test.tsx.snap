// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App component Should match snapshot 1`] = `
<div>
  <div
    class="sc-eCApnc eaZQdw"
  >
    <div
      class="sc-hKFxyN lpoAlx"
    >
      <div
        class="sc-gtsrHT dXGkIp"
      >
        <img
          alt="Harmony logo"
          src="harmony_logo.svg"
        />
        <div
          class="sc-bdnxRM eWoDGl"
        >
          <code
            aria-details="{}"
            aria-label="Balance"
          />
          <code
            aria-details="{}"
            aria-label="Account"
          />
        </div>
      </div>
      <div
        class="sc-dlnjwi GUqgz"
      >
        <code
          aria-details="{}"
          aria-label="InfoContract"
        />
      </div>
    </div>
    <div
      class="Toastify"
    />
  </div>
</div>
`;
